src/Nethermind/Nethermind.Crypto/EthereumEcdsa.cs

```diff
@@ -81,7 +81,8 @@ public bool Verify(Address sender, Transaction tx)
         {
             if (tx.Signature is null)
             {
-                throw new InvalidDataException("Cannot recover sender address from a transaction without a signature.");
+                // TODO: Optimism. No signature in system transactions
+                return Address.Zero;
```

Comment:
Does this not break with non Optimism transactions?

Reply:
I'm not 100% sure, but it should be fine

Reply:
Fixed